receiver/elasticsearchreceiver/internal/model/indexstats.go

```diff
@@ -0,0 +1,30 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/elasticsearchreceiver/internal/model"
+
+// IndexStats represents a response from elasticsearch's /_stats endpoint.
+// The struct is not exhaustive; It does not provide all values returned by elasticsearch,
+// only the ones relevant to the metrics retrieved by the scraper.
+type IndexStats struct {
+	All     IndexStatsIndexInfo            `json:"_all"`
+	Indices map[string]IndexStatsIndexInfo `json:"indices"`
+}
+
+type IndexStatsIndexInfo struct {
+	// TODO: used structs come from nodestats.go, because they refer to the same JSON structures
+	// this should be refactored in a follow-up PR
+	Primaries NodeStatsNodesInfoIndices `json:"primaries"`
+	Total     NodeStatsNodesInfoIndices `json:"total"`
+}
```

Comment:
Most of the structs were already available. I propose that it should be left here as it is and in a follow-up PR I can extract these common structs to another file (one concern: should it be a new, shared file, or should it be this file (`indexstats.go`)?).

Reply:
I think it's ok to just leave it like this. We typically do not merge w/ TODOs in the code, so I think just remove that and we'll call it good. If you want to do a followup PR, I'm happy to review it.